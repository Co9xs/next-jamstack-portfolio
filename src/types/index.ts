export type Category = {
  id: string,
  name: string,
  articleCount?: number
}

export type CategoryData = {
  contents: Category[],
  totalCount: number
}

export type Article = {
  id: string,
  title: string,
  body: string,
  publishedAt: string,
  category: Category | null,
  ogimage?: Image,
  tags?: Tag[],
  author?: Author
  relatedArticles: Article[]
}

export type Image = {
  url: string,
  width: number,
  height: number
}

export type Author = {
  displayName: string,
  description: string,
  image: string
}

export type Tag = {
  id: string,
  name: string
}

export type ArticleData = {
  contents: Article[],
  totalCount: number
}

export type Work = {
  title: string,
  date: string,
  url: string,
  githubUrl: string,
  languages: string[],
  description: string
}

export type Skill = {
  name: string
}

export type Layout = 'Basic' | 'SideBar'