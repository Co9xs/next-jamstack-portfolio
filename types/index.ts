export type Category = {
  name: string
}

export type Article = {
  id: string,
  title: string,
  body: string,
  description: string,
  publishedAt: string,
  category: Category | null
}