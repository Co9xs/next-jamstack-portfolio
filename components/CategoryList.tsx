import styled from 'styled-components'
import { CustomLink } from './CustomLink'
import { CategoryMappedTwemoji } from './CategoryMappedTwemoji'
import { Category } from '@/types'

type Props = {
  categories: Category[]
}

export const CategoryList: React.VFC<Props> = ({ categories }) => {
  return (
    <CategoryListBase>
      {categories.map(category => (
        <CustomLink href={`/blog/categories/${category.id}/page/1`} key={ category.id }>
          <CategoryListItem key={category.id}>
            <CategoryMappedTwemoji category={category}/>
            #{category.name}
            {`(${category.articleCount}記事) `}
          </CategoryListItem>
        </CustomLink>
      ))}
    </CategoryListBase>
  )
}

const CategoryListBase = styled.ul`
  margin: 0;
  padding: 0;
`
const CategoryListItem = styled.li`
  list-style: none;
  margin-bottom: .5rem;
  color: #60A5FA;
  cursor: pointer;
`
