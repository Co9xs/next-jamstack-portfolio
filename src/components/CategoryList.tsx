import styled from 'styled-components'
import { CustomLink } from '@/components'
import { Category } from '@/types'

type Props = {
  categories: Category[]
}

export const CategoryList: React.VFC<Props> = ({ categories }) => {
  return (
    <CategoryListBase>
      <CategoryListHeading>Categories</CategoryListHeading>
      <CategoryListItems>
      {categories.map(category => (
        <CustomLink href={`/blog/categories/${category.id}/page/1`} key={category.id}>
          <CategoryListItem key={category.id}>
            {category.name}
          </CategoryListItem>
        </CustomLink>
      ))}
      </CategoryListItems>
    </CategoryListBase>
  )
}

const CategoryListBase = styled.div`
  margin: 0;
  padding: 0;
`

const CategoryListHeading = styled.h4`
  padding: 0;
  margin-top: 0;
  margin-bottom: 8px;
  font-size: 20px;
  font-weight: bold;
`

const CategoryListItems = styled.ul`
  margin: 0;
  padding: 0;
`

const CategoryListItem = styled.li`
  list-style: none;
  cursor: pointer;
  padding: 8px 0;
  border-bottom: 1px solid #EFEFEF;
  font-size: 18px;
`
