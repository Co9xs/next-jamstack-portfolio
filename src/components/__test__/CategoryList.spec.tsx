import React from 'react'
import { Category } from '@/types';
import renderer,  { act } from 'react-test-renderer'
import { CategoryList } from '../CategoryList';

const mockCategories: Category[] = [
  {
    id: 'tech',
    name: 'Tech'
  },
  {
    id: 'poem',
    name: 'Poem'
  }
]

jest.mock("next/link", () => {
  return ({children}) => {
    return children;
  }
});

jest.mock("react-twemoji", () => {
  return ({children}) => {
    return children;
  }
});

test('CategoryList', () => {
  let component: renderer.ReactTestRenderer
  act(() => {
    component = renderer.create(<CategoryList categories={mockCategories} />)
  })
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})