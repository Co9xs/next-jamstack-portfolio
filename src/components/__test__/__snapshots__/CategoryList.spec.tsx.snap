// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList 1`] = `
<ul
  className="sc-jNMdTA ccJSFx"
>
  <li
    className="sc-dOSReg gvGpFn"
  >
    <span
      className="sc-pFZIQ gYbQTZ"
    >
      ðŸ’»
    </span>
    #
    Tech
    (undefinedè¨˜äº‹) 
  </li>
  <li
    className="sc-dOSReg gvGpFn"
  >
    <span
      className="sc-pFZIQ gYbQTZ"
    >
      ðŸ’¬
    </span>
    #
    Poem
    (undefinedè¨˜äº‹) 
  </li>
</ul>
`;
